build.sh

```diff
@@ -395,7 +395,16 @@ if [[ "$sourceOnly" == "true" ]]; then
     }
 
     GIT_DIR="$scriptroot/.git"
-    if [ -f "$GIT_DIR/index" ]; then # We check for index because if outside of git, we create config and HEAD manually
+    # Check if exist .git file
+    if [ -f "$GIT_DIR" ]; then
+      if ! grep -iq '^gitdir: ' "$GIT_DIR"; then
+        echo "ERROR: $GIT_DIR exist, but does not contain a valid 'gitdir:' pointer."
+        exit 1
+      else
+        GIT_DIR=$(grep -oP '^gitdir: \K.*' "$GIT_DIR")
```

Comment:
Perl regex feature is not supported on the BusyBox version of grep which is what is used by Alpine Linux. This script needs to support Alpine.

Reply:
Update: https://github.com/dotnet/dotnet/pull/1874/commits/8c0e4c1a3356df896a1af403d547ba4906f8926b